core/src/main/java/org/apache/calcite/rel/type/RelRecordType.java

```diff
@@ -29,19 +29,29 @@
 public class RelRecordType extends RelDataTypeImpl implements Serializable {
   /** Name resolution policy; usually {@link StructKind#FULLY_QUALIFIED}. */
   private final StructKind kind;
+  private final boolean nullable;
 
   //~ Constructors -----------------------------------------------------------
 
   /**
    * Creates a <code>RecordType</code>. This should only be called from a
    * factory method.
    */
-  public RelRecordType(StructKind kind, List<RelDataTypeField> fields) {
+  public RelRecordType(StructKind kind, List<RelDataTypeField> fields, boolean nullable) {
```

Comment:
Minor: maybe it would be better to add the new constructor after the old one. Usually the telescoping constructor pattern puts constructor in increasing argument order.

Reply:
I agree with your logic, but I just kept the existing one: the class already defined the 2-parameter constructor before the 1-parameter constructor, so I just added the 3-parameter one before both of them. I could re-arrange the constructors into an increasing argument order, though.

Reply:
No don't bother, I didn't see that there were already two constructors. Maybe just add a bit of Javadoc.